src/aggregate/aggregate_exec.c

```diff
@@ -293,9 +305,14 @@ static int execCommandCommon(RedisModuleCtx *ctx, RedisModuleString **argv, int
     if (rc != REDISMODULE_OK) {
       goto error;
     }
+  } else if (IsProfile(r)) {
+    RedisModule_ReplyWithArray(ctx, 2);
+    sendChunk(r, ctx, -1);
+    Profile_Print(ctx, r);
+    AREQ_Free(r);
   } else {
-    // Execute() will call free when appropriate.
-    AREQ_Execute(r, ctx);
+    sendChunk(r, ctx, -1);
+    AREQ_Free(r);
```

Comment:
Can move `AREQ_Free(r);` outside of if/else blocks.

Reply:
It isn't that simple since the cursor is part of this if-else flow and may not require `AREQ_Free`.